src/metrics.cpp

```diff
@@ -225,12 +268,25 @@ int printStats(bool mining)
 
     if (IsInitialBlockDownload(Params())) {
         int netheight = currentHeadersHeight == -1 || currentHeadersTime == 0 ? 
-            0 : EstimateNetHeight(Params().GetConsensus(), currentHeadersHeight, currentHeadersTime);
+            0 : EstimateNetHeight(params, currentHeadersHeight, currentHeadersTime);
         int downloadPercent = height * 100 / netheight;
         std::cout << "     " << _("Downloading blocks") << " | " << height << " / ~" << netheight << " (" << downloadPercent << "%)" << std::endl;
     } else {
         std::cout << "           " << _("Block height") << " | " << height << std::endl;
     }
+
+    auto secondsLeft = SecondsLeftToNextEpoch(params, height);
+    std::string strUpgradeTime;
+    if (secondsLeft == boost::none) {
```

Comment:
If the order of the conditional branches is reversed, then this can simply be:
```suggestion
    if (secondsLeft) {
```

Reply:
Done at https://github.com/zcash/zcash/commit/9582e258ba7b977cfb035aeadf07c80d7a8f7de7